lib/src/sv_sign.c

```diff
@@ -923,12 +918,15 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
     // Temporally turn the PEM |private_key| into an EVP_PKEY and allocate memory for signatures.
     SV_THROW(openssl_private_key_malloc(self->sign_data, private_key, private_key_size));
     SV_THROW(openssl_read_pubkey_from_private_key(self->sign_data, &self->pem_public_key));
+    self->private_key = private_key;
 
-    // Fallback to signing plugin if ONVIF is unavailable or ONVIF cant be initialized.
-    if(!self->onvif || !(initialize_onvif(self) == SV_OK)) {
+    // Try to initialize ONVIF; if it is not available or cannot be initialized,
+    // fallback to Signed Video signing plugin.
+    initialize_onvif(self);
+    if (!self->onvif) {
```

Comment:
Now when ONVIF signing plugin is called through the Axis signing plugin (see other PR) you need to remove the if-statement